components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2453,6 +2453,17 @@ public String getSwagger12Definition(APIIdentifier apiId) throws APIManagementEx
 		return apiJSON.toJSONString();
 	}
 
+    /**
+     * Returns the all the Consumer keys of applications which are subscribed to the given API
+     *
+     * @param apiIdentifier APIIdentifier
+     * @return a String array of ConsumerKeys
+     * @throws APIManagementException
+     */
+    public String[] getConsumerKeys(APIIdentifier apiIdentifier) throws APIManagementException {
+
+        return apiMgtDAO.getConsumerKeys(apiIdentifier);
```

Comment:
Why do we expose DAOs to HostObjects and other entities? Do we have any other "API" which wraps the DAO logics and expose some business functionalities which could have been used to replace this line?